source/agora/node/FullNode.d

```diff
@@ -149,6 +157,18 @@ public class FullNode : API
             this.storage, this.enroll_man, this.pool, onValidatorsChanged);
         this.exception = new RestException(
             400, Json("The query was incorrect"), string.init, int.init);
+
+        // Make `BlockExternalizedHandler` from config
+        foreach (block_externalize_address;
+            config.event_handlers.block_externalized_handler_addresses)
+            this.block_handlers[block_externalize_address] = this.network
+                .getBlockExternalizedHandler(block_externalize_address);
+
+        // Make `PreImageReceivedHandler` from config
+        foreach (preimage_updated_address;
+            config.event_handlers.preimage_updated_handler_addresses)
+            this.preimage_handlers[preimage_updated_address] = this.network
+                .getPreimageReceivedHandler(preimage_updated_address);
```

Comment:
This can be simplified because the variables are not re-used outside the foreach statement.

```suggestion
        // Make `BlockExternalizedHandler` from config
        foreach (address;
            config.event_handlers.block_externalized_handler_addresses)
            this.block_handlers[address] = this.network.getBlockExternalizedHandler(address);

        // Make `PreImageReceivedHandler` from config
        foreach (address;
            config.event_handlers.preimage_updated_handler_addresses)
            this.preimage_handlers[address] = this.network.getPreimageReceivedHandler(address);
```

Reply:
Good! The code is more concise.